packages/swapper/src/swappers/RelaySwapper/utils/getTrade.ts

```diff
@@ -90,15 +90,60 @@ export async function getTrade<T extends 'quote' | 'rate'>({
 
   if (maybeQuote.isErr()) return Err(maybeQuote.unwrapErr())
 
-  const quote = maybeQuote.unwrap()
+  const { data: quote } = maybeQuote.unwrap()
 
-  const swapSteps = quote.data.steps.filter(step => step.id !== 'approve')
-  const hasApprovalStep = quote.data.steps.find(step => step.id === 'approve')
+  if (!quote.details) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay quote details not found',
+      }),
+    )
+  }
+
+  const { slippageTolerance, rate, currencyOut, timeEstimate } = quote.details
+
+  if (!slippageTolerance) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay quote slippage tolerance not found',
+      }),
+    )
+  }
+
+  if (!rate) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay quote rate not found',
+      }),
+    )
+  }
 
-  if (swapSteps.length > 2) {
-    deps.mixPanel?.track(MixPanelEvent.RelayMultiSteps, {
+  if (!currencyOut) {
+    return Err(
+      makeSwapErrorRight({
+        message: 'Relay quote relayToken not found',
+      }),
```

Comment:
This isn't really user-facing, I don't think there's value in the 3 being explicit monadic errors.

Reply:
They are all gone in [29fc44d](https://github.com/shapeshift/web/pull/9188/commits/29fc44d8ce158086f2cbec2c3c7ee0dd44302dac)